### PR TITLE
Fix to use prepared statements to prevent SQL injection attacks

### PR DESCRIPTION
**Pull request by chriscrossman-ibm**

### CODE REVIEW

File: src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java [hunk 1]
```diff
@@ -85,13 +85,13 @@ public void createTable(Connection connection, String tableName, Schema schema)
...
         }
 
-        final String createQuery = String.format(CREATE_STATEMENT, tableName, String.join(", ", fieldDatabaseDefinitions));
-        logger.info("CREATEQuery: " + createQuery);
-
-        Statement statement = connection.createStatement();
-        statement.execute(createQuery);
+        PreparedStatement pstmt = connection.prepareStatement(CREATE_STATEMENT);
+        pstmt.setString(1, tableName);
+        pstmt.setString(1, String.join(", ", fieldDatabaseDefinitions));
```
Comment (iamgollum) [inline src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java:90]: need index 2 here

File: src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java [hunk 2]
```diff
@@ -135,13 +135,16 @@ public void insert(String tableName, Collection<SinkRecord> records) throws SQLE
                 String listTableFields = String.join(", ", fieldNames);
                 String listDataFields = String.join(", ", fieldValues);
 
-                final String finalQuery = String.format(INSERT_STATEMENT, tableName, listTableFields, listDataFields);
-                statement.addBatch(finalQuery);
-                logger.debug("Final prepared statement: '{}' //", finalQuery);
+                pstmt.setString(1, tableName);
+                pstmt.setString(1, listTableFields);
```
Comment (iamgollum) [inline src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java:139]: index 2

File: src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java [hunk 3]
```diff
@@ -135,13 +135,16 @@ public void insert(String tableName, Collection<SinkRecord> records) throws SQLE
...
+                pstmt.setString(1, listDataFields);
```
Comment (iamgollum) [inline src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java:140]: index 3

**Review by iamgollum — CHANGES_REQUESTED** [review]

**Comment by iamgollum** [comment]
@chriscrossman-ibm make sure to test this guy. I know unit testing is hard for this without JDBC ORM wrappers, etc

File: src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java (hunk 2 above)
Comment (chriscrossman-ibm) [reply src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java]: Fixed

File: src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java (hunk 3 above)
Comment (chriscrossman-ibm) [reply src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java]: Fixed

File: src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java (hunk 1 above)
Comment (chriscrossman-ibm) [reply src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java]: Fixed

**Review by iamgollum — APPROVED** [review]

**Pull request merged**